### PR TITLE
Fix linter warnings

### PR DESCRIPTION
**Pull request by vspetrov**
Actual issues are fixed, unused vars are suppressed with NOLINT

### CODE REVIEW

File: src/coll_score/ucc_coll_score.c [hunk 1]
```diff
@@ -414,7 +414,7 @@ static ucc_status_t str_to_msgranges(const char *str, size_t **ranges,
 
 static ucc_status_t ucc_coll_score_parse_str(const char       *str,
                                              ucc_coll_score_t *score,
-                                             ucc_rank_t        team_size)
+                                             ucc_rank_t        team_size)//NOLINT
```
Comment (manjugv) [inline src/coll_score/ucc_coll_score.c:417]: @vspetrov Adding NOLINT directive to the code seems like distraction? Is there a different way that is less distractive to add the directives?

Comment (vspetrov) [reply src/coll_score/ucc_coll_score.c:417]: I dont know another way. Those are "unused vars" in the functions that are not fully implemented. We either need to suppress these false positives, OR remove check for unused vars from Linter. @manjugv @Sergei-Lebedev thoughts?

File: src/components/tl/ucp/tl_ucp_context.c [hunk 2]
```diff
@@ -136,7 +136,7 @@ UCC_CLASS_INIT_FUNC(ucc_tl_ucp_context_t,
 static void ucc_tl_ucp_context_barrier(ucc_tl_ucp_context_t *ctx,
                                        ucc_context_oob_coll_t *oob)
 {
-    char        *rbuf = ucc_malloc(sizeof(char*) * oob->participants,
+    char        *rbuf = ucc_malloc(sizeof(char) * oob->participants,
```
Comment (Sergei-Lebedev) [inline src/components/tl/ucp/tl_ucp_context.c:139]: also fix tl_error sizeof(char*) * oob->participants if failed to allocate rbuf

**Review by Sergei-Lebedev — APPROVED** [review]

File: src/coll_score/ucc_coll_score.c (hunk 1 above)
Comment (Sergei-Lebedev) [reply src/coll_score/ucc_coll_score.c:417]: I think NOLINT here is  just another reminder that something is not implemented, so it looks fine to me

**Comment by vspetrov** [comment]
@manjugv @bureddy plz have a look

**Review by bureddy — APPROVED** [review]

**Pull request merged**